lib/client.js

```diff
@@ -874,18 +1013,26 @@ MqttClient.prototype._handlePingresp = function () {
  */
 
 MqttClient.prototype._handleConnack = function (packet) {
-  var rc = packet.returnCode
-  var errors = [
-    '',
-    'Unacceptable protocol version',
-    'Identifier rejected',
-    'Server unavailable',
-    'Bad username or password',
-    'Not authorized'
-  ]
+  var version = this.options.protocolVersion
+  var rc = version === 5 ? packet.reasonCode : packet.returnCode
 
   clearTimeout(this.connackTimer)
 
+  if (packet.properties) {
+    if (packet.properties.topicAliasMaximum) {
+      if (!this.options.properties) { this.options.properties = {} }
+      this.options.properties.topicAliasMaximum = packet.properties.topicAliasMaximum
+    }
+    if (packet.properties.serverKeepAlive && this.options.keepalive) {
+      this.options.keepalive = packet.properties.serverKeepAlive
+      this._shiftPingInterval()
+    }
+    if (packet.properties.maximumPacketSize) {
+      if (!this.options.properties) { this.options.properties = {} }
+      this.options.properties.maximumPacketSize = packet.properties.maximumPacketSize
+    }
+  }
```

Comment:
Can you extract `var options = this.options`?

Reply:
fixed